### ISSUE TITLE
issue bulding captagent on CentOS: undefined reference to `rpl_malloc'

### ISSUE THREAD

**Issue by fborot** [sipcapture/captagent#57]
when trying to build captagent I get this error when doing "make"

`[root@AW-MIA-PROXY-02 captagent]# make
Making all in src
make[1]: Entering directory `/usr/src/captagent/src'
make  all-recursive
make[2]: Entering directory `/usr/src/captagent/src'
Making all in .
make[3]: Entering directory `/usr/src/captagent/src'
/bin/sh ../libtool  --tag=CC   --mode=link gcc -g -fPIC -rdynamic -I../include -g -O2   -o captagent captagent.o conf_function.o log.o md5.o modules.o xmlread.o capplan.o capplan.tab.o    -lfl -lpcap -lexpat -ldl -lpthread
libtool: link: gcc -g -fPIC -rdynamic -I../include -g -O2 -o captagent captagent.o conf_function.o log.o md5.o modules.o xmlread.o capplan.o capplan.tab.o  -lfl -lpcap /usr/lib64/libexpat.so -ldl -lpthread -Wl,-rpath -Wl,/usr/lib64 -Wl,-rpath -Wl,/usr/lib64
capplan.tab.o: In function `yyparse':
/usr/src/captagent/src/capplan.tab.c:1362: undefined reference to `rpl_malloc'
collect2: ld returned 1 exit status
make[3]: *** [captagent] Error 1
make[3]: Leaving directory `/usr/src/captagent/src'
make[2]: *** [all-recursive] Error 1
make[2]: Leaving directory `/usr/src/captagent/src'
make[1]: *** [all] Error 2
make[1]: Leaving directory `/usr/src/captagent/src'
make: *** [all-recursive] Error 1
[root@AW-MIA-PROXY-02 captagent]#`

these are the version of my libraries/dependencies:
autoconf-2.69
automake-1.15
flex-2.5.36
json-c-0.10
libtool-2.4.6
libpcap-1.1.1

what can it be? thank you in advance

**Comment by fborot** [comment]
I this link with a solution to the undefined reference:
[https://groups.google.com/forum/#!topic/ikarus-users/_R0QHqwyYz8](url)

If you get a problem about an undefined reference to rpl_malloc, you
probably need to do:
$ export ac_cv_func_malloc_0_nonnull=yes
before running ./configure ...

but now I get this other error:

`make[4]: Entering directory `/usr/src/captagent/src/modules/transport/json'
/bin/sh ../../../../libtool  --tag=CC   --mode=compile gcc -DHAVE_CONFIG_H -I. -I../../../../src   -I../../../../include  -Wall  -g -O2 -MT transport_json_la-transport_json.lo -MD -MP -MF .deps/transport_json_la-transport_json.Tpo -c -o transport_json_la-transport_json.lo `test -f 'transport_json.c' || echo './'`transport_json.c
libtool: compile:  gcc -DHAVE_CONFIG_H -I. -I../../../../src -I../../../../include -Wall -g -O2 -MT transport_json_la-transport_json.lo -MD -MP -MF .deps/transport_json_la-transport_json.Tpo -c transport_json.c  -fPIC -DPIC -o .libs/transport_json_la-transport_json.o
In file included from transport_json.c:44:
/usr/local/include/json/json.h:27:34: error: json_object_iterator.h: No such file or directory
transport_json.c: In function âsend_jsonâ:
transport_json.c:264: warning: passing argument 1 of âsend_dataâ discards qualifiers from pointer target type
transport_json.c: In function âsigPipeâ:
transport_json.c:545: warning: assignment from incompatible pointer type
make[4]: *** [transport_json_la-transport_json.lo] Error 1
make[4]: Leaving directory `/usr/src/captagent/src/modules/transport/json'
make[3]: *** [all-recursive] Error 1
make[3]: Leaving directory `/usr/src/captagent/src/modules/transport/json'
make[2]: *** [all-recursive] Error 1
make[2]: Leaving directory `/usr/src/captagent/src'
make[1]: *** [all] Error 2
make[1]: Leaving directory `/usr/src/captagent/src'
make: *** [all-recursive] Error 1
[root@AW-MIA-PROXY-02 captagent]# `

**Comment by lmangani** [comment]
Please provide the full OS details/version and the full process you are following before running ```make```

**Comment by fborot** [comment]
OS version:
CentOS release 5.8 (Final)
Linux AW-MIA-PROXY-02 2.6.18-308.el5 #1 SMP Tue Feb 21 20:06:06 EST 2012 x86_64 x86_64 x86_64 GNU/Linux

The steps are the same listed in the wiki:
  cd /usr/src
  git clone https://github.com/sipcapture/captagent.git captagent
  cd captagent
  ./build.sh
  ./configure
  make && make install

the probs that I have were due to issues with he dependencies, for example the autoconfig present in the machine was too old, and with yum I could not get a newer one so I had to download the sources and build again. that happened with automake, libtool, libpcap and json-c

I applied the step shown below and I was able to make & make install:

https://github.com/json-c/json-c/issues/35
Line 27 of /usr/local/include/json/json.h defined json_object_iterator.h to be included, but that file is missing in the include directory and has to be manually copied over from the source directory.

I am going to replicate my steps in another machine to verify that there is no prob just in case.
